File: nunchuk-widget/src/main/java/com/nunchuk/android/widget/NCFontTabLayout.java
package com.nunchuk.android.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;

public class NCFontTabLayout extends TabLayout {
    public NCFontTabLayout(final Context context) {
        super(context);
    }

    public NCFontTabLayout(final Context context, final AttributeSet attrs) {
        super(context, attrs);
    }

    public NCFontTabLayout(final Context context, final AttributeSet attrs, final int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void setupWithViewPager(final ViewPager viewPager) {
        super.setupWithViewPager(viewPager);
        customizeTabs(viewPager);
    }

    public void customizeTabs(@NonNull final ViewPager viewPager) {
        removeAllTabs();

        final ViewGroup slidingTabStrip = (ViewGroup) getChildAt(0);
        final PagerAdapter adapter = viewPager.getAdapter();

        final int space = (int) getContext().getResources().getDimension(R.dimen.nc_padding_8);
        for (int i = 0, count = adapter.getCount(); i < count; i++) {
            final Tab tab = newTab();
            addTab(tab.setText(adapter.getPageTitle(i)));
            final AppCompatTextView view = (AppCompatTextView) ((ViewGroup) slidingTabStrip.getChildAt(i)).getChildAt(1);
            view.setTypeface(FontStyle.getCurrentTypeface(view.getContext(), FontStyle.STYLE_SEMI_BOLD));
            view.measure(MeasureSpec.UNSPECIFIED, MeasureSpec.UNSPECIFIED);
            view.setLayoutParams(new LinearLayout.LayoutParams(view.getMeasuredWidth() + space, LayoutParams.WRAP_CONTENT));
        }
    }

}

File: nunchuk-widget/src/main/java/com/nunchuk/android/widget/NCFontTabLayout.kt
/**************************************************************************
 * This file is part of the Nunchuk software (https://nunchuk.io/)        *
 * Copyright (C) 2022, 2023 Nunchuk                                       *
 *                                                                        *
 * This program is free software; you can redistribute it and/or          *
 * modify it under the terms of the GNU General Public License            *
 * as published by the Free Software Foundation; either version 3         *
 * of the License, or (at your option) any later version.                 *
 *                                                                        *
 * This program is distributed in the hope that it will be useful,        *
 * but WITHOUT ANY WARRANTY; without even the implied warranty of         *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          *
 * GNU General Public License for more details.                           *
 *                                                                        *
 * You should have received a copy of the GNU General Public License      *
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.  *
 *                                                                        *
 **************************************************************************/

package com.nunchuk.android.widget

import android.content.Context
import android.util.AttributeSet
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatTextView
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.nunchuk.android.widget.util.FontInitializer

class NCFontTabLayout @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : TabLayout(context, attrs, defStyleAttr) {
    private val fontStyle = FontInitializer(context)

    override fun setupWithViewPager(viewPager: ViewPager?) {
        super.setupWithViewPager(viewPager)
        viewPager?.let { customizeTabs(it) }
    }

    private fun customizeTabs(viewPager: ViewPager) {
        removeAllTabs()
        val slidingTabStrip = getChildAt(0) as ViewGroup
        val adapter = viewPager.adapter
        var i = 0
        val count = adapter!!.count
        while (i < count) {
            val tab = newTab()
            addTab(tab.setText(adapter.getPageTitle(i)))
            val view = (slidingTabStrip.getChildAt(i) as ViewGroup).getChildAt(1) as AppCompatTextView
            view.typeface = fontStyle.getCurrentTypeface(FontInitializer.STYLE_SEMI_BOLD)
            view.measure(MeasureSpec.UNSPECIFIED, MeasureSpec.UNSPECIFIED)
            i++
        }
    }
}